test/test_storage.py

```diff
@@ -137,11 +139,15 @@ def progress_callback(pos, total):
         # sub3 is a file. Actual object storage systems support this, but a file system does not
         with pytest.raises(NotADirectoryError):
             st.store_file_from_memory("test1/sub3/sub3.1/sub3.1.1", b"1", None)
+    elif driver == 'sftp':
+        # sub3 is a file. Actual object storage systems support this, but a file system does not
+        with pytest.raises(errors.StorageError):
+            st.store_file_from_memory("test1/sub3/sub3.1/sub3.1.1", b"1", None)
     else:
         st.store_file_from_memory("test1/sub3/sub3.1/sub3.1.1", b"1", None)
         created_keys.add("test1/sub3/sub3.1/sub3.1.1")
 
-    if driver == "local":
+    if driver in ("local", "sftp"):
```

Comment:
Maybe list or set instead (same for other similar checks)? Tuples are nasty when you remove all but the last element and don't remember the extra comma.